framework/src/main/java/org/apache/drill/test/framework/TestVerifier.java

```diff
@@ -348,16 +349,38 @@ private static int getMissingCount(Map<ColumnList, Integer> map) {
   }
 
   private static int compareTo(ColumnList list1, ColumnList list2,
-      Map<Integer, String> columnIndexAndOrder, int start) {
+      List<IndexAndOrder> columnIndexAndOrder, int start)
+       throws VerificationException {
+    return compareTo(list1, list2, columnIndexAndOrder, start, null);
+  }
+
+  /**
+   * Compare two rows, represented by list1 and list2.
+   * Compare the two rows in the order specified in the order by clause.  If the
+   * two rows have have different values in the first order-by column, then it
+   * can be determined if they are in the right order without looking at the
+   * second order-by column.  If the two rows have the same value in the first
+   * order-by column, then the second order-by column is checked.  The second
+   * order-by column (and subsequent order-by columns) are handled in a recursive
+   * call to this procedure.
+   * Numerical values are handled separately from other values.
+   * JSON strings have special handling because they have to be parsed.  A field
+   * within a JSON string can be used in an order-by clause, so the field needs
+   * to be extracted from the JSON string.
+   * This code currently handles integer and string values in JSON strings.
+   */
+  private static int compareTo(ColumnList list1, ColumnList list2,
```

Comment:
@rhou1 This part of the code is never reached (code has been abandoned I believe). Let's seat down and take a closer look.